kroxylicious-api/src/main/java/io/kroxylicious/proxy/metadata/ResourceMetadataRequest.java

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.metadata;
+
+public sealed interface ResourceMetadataRequest<R extends ResourceMetadataResponse<?>> permits DescribeTopicLabelsRequest, ListTopicsRequest {
```

Comment:
Entirely for my own education what is the benefit of sealing the interface and explicitly listing the implementations?

Reply:
Two things, neither of which are _essential_, but which don't appear to be harmful:

* It makes it very clear that the implementations are owned by the runtime, not plugin authors. So the code is more self-documenting.
* It will allow us to `switch` over the types of request. This is hardly compelling because we expect to only need to do this in one place, the handler.